src/app/parcels/fetchParcelTableData.ts

```diff
@@ -322,3 +323,21 @@ export const fetchParcelStatuses = async (): Promise<ParcelStatusesReturnType> =
 
     return { data: parcelStatusesList, error: null };
 };
+
+export const getParcelsWithEvent = async (
+    targetEventName: StatusType,
+    parcelIds: string[]
+): Promise<GetDbParcelDataResult> => {
+    const { data, error } = await supabase
+        .from("parcels_plus")
+        .select("*, events!inner(new_parcel_status)")
```

Comment:
Would this work if a parcel has no events associated with it?

Reply:
I think so yeah. The inner join means only parcels with events are returned, which is all we want here